### PR TITLE
Fix download all endpoint

### PR DESCRIPTION
**Pull request by olaughter**
# Description

Fixes the download all query, and adds a test.

This query broke following schema changes. This updates the sql to work for the new schema. It also adds an initial test that checks endpoint successfully returns a redirect url. This makes sure the endpoint works. Data validation would be a sensible next step but is beyond the scope here.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [x] The PR represents a single feature (small driveby fixes are also ok)
- [x] The PR includes tests that are sufficient for the level of risk
- [x] The code is sufficiently commented, particularly in hard-to-understand areas
- [x] Any required documentation updates have been made
- [x] Any TODOs added are captured in future tickets
- [x] No FIXMEs remain

### CODE REVIEW

File: app/core/download.py [hunk 1]
```diff
@@ -189,10 +189,12 @@ def create_query(ingest_cycle_start: str) -> str:
         "    ON d.family_import_id = f.import_id "
         "inner join geography g "
         "    on g.id = f.geography_id "
-        "join family_organisation fo "
-        "    on fo.family_import_id = f.import_id "
+        "join family_corpus fc "
```
Comment (THOR300) [inline app/core/download.py:192]: I opened this file to read the surrounding SQL and it's rather complex and long! 

Do you think this should be managed in a different way?

**Review by THOR300 — APPROVED** [review]
LGTM - just the query re the sql!

File: app/core/download.py (hunk 1 above)
Comment (olaughter) [reply app/core/download.py:192]: Maybe, but not as part of this change

Comment (olaughter) [reply app/core/download.py:192]: I'm not sure there is an ideal solution for this complex of a query, like we could move it to use sqlalchemy's ORM, but it's likely to be messier and harder to read ¯\_(ツ)_/¯

Comment (diversemix) [reply app/core/download.py:192]: Half way house may be to pop it in a file and read it in - so we can test it manual if needs be /cc @katybaulch

Comment (katybaulch) [reply app/core/download.py:192]: Yeah, Peter's approach is likely best.

**Pull request merged**